web/src/components/shared/AirgapRegistrySettings.jsx

```diff
@@ -58,9 +61,9 @@ class AirgapRegistrySettings extends Component {
       await this.props.updateRegistryDetails({ appSlug, hostname, username, password, namespace });
       await this.props.getKotsAppRegistryQuery.refetch();
 
-      const interval = setInterval(this.updateStatus.bind(this), 1000);
+      const cancel = new Repeater(this.updateStatus, 1000).Start();
```

Comment:
why not just return the object and use it instead of returning the cancel function and calling it?

Reply:
something like `const repeater = new Repeater(this.updateStatus, 1000)` then `repeater.start` and `repeater.cancel`

Reply:
and save the repeater in the state

Reply:
no reason